invoices/views.py

```diff
@@ -119,3 +119,25 @@ def new_comment(request, invoice_pk):
 
     return HttpResponseRedirect(reverse('invoices-show', kwargs={'pk': invoice_pk}))
 
+@require_http_methods(["GET", "POST"])
+@login_required
+def delete_invoice(request, pk):
+    """
+    Delete invoice. Ask for confirmation on GET and send to POST.
+    """
+    try:
+        invoice = Invoice.objects.get(pk=pk)
+    except ObjectDoesNotExist:
+        raise Http404("Fakturan finns inte")
+
+    if not request.user.profile.may_delete_invoice(invoice):
+        return HttpResponseForbidden('Du har inte behörighet att ta bort denna faktura.')
+    if invoice.is_payed():
```

Comment:
This branch is unreachable. Since `user.may_delete_invoice(invoice)` also does this check, the first condition of this if-statement will always return `false` if the invoice is paid. It would be sensible not to do this check twice, since even though it has no effect now, someone may miss one of them in the future when (if) the system is being reworked. I leave it up to you to decide where to do this, but it should only be done once. 

Also, past tense of 'pay' is 'paid' :)

Reply:
Fun fact: `grep -r "payed" . | wc -l` returnerar `40`